src/App.tsx

```diff
@@ -1,18 +1,61 @@
 import React from 'react';
 import './App.css';
-import { getNumbers } from './utils';
-
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42)
-  .map(n => `Item ${n}`);
+import {
+  Route, Routes, useNavigate, useSearchParams,
+} from 'react-router-dom';
+import { Pagination } from './components/Pagination';
+import { items } from './items';
+import { getSearchWith } from './searchHelper';
 
 export const App: React.FC = () => {
+  const [searchParams] = useSearchParams();
+  const navigate = useNavigate();
+
+  const handleOnSelect = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    let newSearchParams = getSearchWith(searchParams, 'page', 1);
+
+    newSearchParams = getSearchWith(
+      newSearchParams,
+      'perPage',
+      +e.target.value,
+    );
+
+    navigate(`?${newSearchParams}`);
+  };
+
+  const page = searchParams.get('page') !== null
+    ? Number(searchParams.get('page'))
+    : 1;
+  const perPage = searchParams.get('perPage') !== null
+    ? Number(searchParams.get('perPage'))
+    : 5;
+
+  const firstDisplayedItem = page * perPage - perPage + 1;
+  const lastDisplayedItem = page * perPage > items.length
+    ? items.length
+    : page * perPage;
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        Page
+        {' '}
```

Comment:
it's a bad practice for adding spaces this way, it's better to use paddings and margins for this